core/domain/summary_services.py

```diff
@@ -187,7 +201,7 @@ def _generate_query(categories):
         summary_dicts = [
             all_summary_dicts[exp_id] for exp_id in exp_ids_to_display
             if exp_id in all_summary_dicts]
-
+        summary_dicts = sorted(summary_dicts, key=lambda k: k['created_on'])
```

Comment:
Sure. But in this case, I'm just arranging the summary_dicts by creation date and all the groups including "recently published" gets arranged accordingly and limit remains same, 8.